### PR TITLE
chore: enable KongCustomEntity feature gate by default

### LINKED ISSUE #6124: Enable `KongCustomEntity` by default
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does this enhancement require public documentation?

- [X] I have added an Acceptance Criteria item for adding and/or adjusting public documentation (if applicable)

### Problem Statement

`KongCustomEntity` feature gate is imported to enable the controller to reconcile `KongCustomEntity` CRs for configuring custom Kong entities. It is disabled by default when the feature first imported. We need to enable it by default when the feature is ready for beta/GA.

### Proposed Solution

- Flip `KongCustomEntity` feature gate to be enabled by default.

### Additional information

_No response_

### Acceptance Criteria

- [x]  `KongCustomEntity`  is enabled by default
- [x] `FEATURE_GATES.md` and corresponding docs page include introduction of the feature gate

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:

enable custom entities by default. `KongCustomEntity` feature gate is enabled by default and promoted to beta.

**Which issue this PR fixes**:

fixes #6124.

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: internal/dataplane/testdata/golden/fallback-config-kong-custom-entities-ee/default_golden.yaml
Comment (czeslavo) [inline internal/dataplane/testdata/golden/fallback-config-kong-custom-entities-ee/default_golden.yaml:1]: Please remove `testdata/golden/fallback-config-kong-custom-entities-ee/custom-entities-on_*.yaml` as it is now an equivalent of the default settings.

File: internal/dataplane/testdata/golden/kong-custom-entity-ee/default_golden.yaml
Comment (czeslavo) [inline internal/dataplane/testdata/golden/kong-custom-entity-ee/default_golden.yaml:1]: Ditto. Please remove `testdata/golden/kong-custom-entity-ee/custom-entities-on_*.yaml`.

File: internal/dataplane/testdata/golden/fallback-config-kong-custom-entities-ee/default_golden.yaml
Comment (randmonkey) [reply internal/dataplane/testdata/golden/fallback-config-kong-custom-entities-ee/default_golden.yaml:1]: Done.

File: internal/dataplane/testdata/golden/kong-custom-entity-ee/default_golden.yaml
Comment (randmonkey) [reply internal/dataplane/testdata/golden/kong-custom-entity-ee/default_golden.yaml:1]: Done.

Comment (czeslavo) [reply internal/dataplane/testdata/golden/kong-custom-entity-ee/default_golden.yaml:1]: Have you pushed the changes? I can only see `testdata/golden/fallback-config-kong-custom-entities-ee/custom-entities-on_*.yaml` removed, but `testdata/golden/kong-custom-entity-ee/custom-entities-on_*.yaml` remain.

Comment (randmonkey) [reply internal/dataplane/testdata/golden/kong-custom-entity-ee/default_golden.yaml:1]: Removed them now.

**Review by czeslavo — APPROVED** [review]

**Pull request merged**